tracer/test/Datadog.Trace.IntegrationTests/Tagging/AASTagsTests.cs

```diff
@@ -0,0 +1,249 @@
+        span.GetTag(Tags.AzureAppServicesResourceId).Should().Be("/subscriptions/subscriptionid/resourcegroups/siteresourcegroup/providers/microsoft.web/sites/sitename");
+        span.GetTag(Tags.AzureAppServicesInstanceId).Should().Be("InstanceId");
+        span.GetTag(Tags.AzureAppServicesInstanceName).Should().Be("InstanceName");
+        span.GetTag(Tags.AzureAppServicesOperatingSystem).Should().Be("windows");
+        span.GetTag(Tags.AzureAppServicesRuntime).Should().Be(FrameworkDescription.Instance.Name);
+        span.GetTag(Tags.AzureAppServicesExtensionVersion).Should().Be("unknown");
+
+        return true;
+    }
+
+    private bool AssertNonRootSpan(MockSpan span)
+    {
+        span.GetTag(Tags.AzureAppServicesSiteName).Should().Be("SiteName");
+        span.GetTag(Tags.AzureAppServicesSiteType).Should().Be("app");
+        span.GetTag(Tags.AzureAppServicesSiteKind).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesResourceGroup).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesSubscriptionId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesResourceId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesInstanceId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesInstanceName).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesOperatingSystem).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesRuntime).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesExtensionVersion).Should().BeNull();
+
+        return true;
+    }
+
+    private bool AssertNoTags(MockSpan span)
+    {
+        span.GetTag(Tags.AzureAppServicesSiteName).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesSiteType).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesSiteKind).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesResourceGroup).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesSubscriptionId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesResourceId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesInstanceId).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesInstanceName).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesOperatingSystem).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesRuntime).Should().BeNull();
+        span.GetTag(Tags.AzureAppServicesExtensionVersion).Should().BeNull();
+
+        return true;
+    }
+
+    private IConfigurationSource GetNullMockVariables()
+    {
+        var vars = Environment.GetEnvironmentVariables();
+
+        var collection = new NameValueCollection();
+        foreach (DictionaryEntry kvp in vars)
+        {
+            collection.Add(kvp.Key as string, kvp.Value as string);
+        }
+
+        return new NameValueConfigurationSource(collection);
+    }
+
+    private IConfigurationSource GetMockVariables()
+    {
+        var vars = Environment.GetEnvironmentVariables();
```

Comment:
Same, this would be simpler if you just didn't load the ambient env vars, and only set the ones you are testing

Reply:
Fixed in ef90cd7338723f667f6d6d3e37422dc9da716f41